tests/Microsoft.Identity.Test.Unit/PublicApiTests/LoggerTests.cs

```diff
@@ -23,33 +29,51 @@ public void TestInit()
+            if (legacyLogger)
+            {
+                return new LegacyIdentityLoggerAdapter(Guid.Empty, null, null, logLevel, enablePiiLogging, true, _callback);
+            }
+
+            return new IdentityLoggerAdapter(new TestIdentityLogger(LoggerAdapterHelper.GetEventLogLevel(logLevel)), Guid.Empty, null, null, enablePiiLogging);
+        }
+
+        [TestMethod()]
+        public void IdentityLoggerConstructorComponentTest()
+        {
+            ILoggerAdapter logger = new IdentityLoggerAdapter(null, Guid.Empty, "", "", false);
+            Assert.AreEqual(string.Empty, logger.ClientName);
+            Assert.AreEqual(string.Empty, logger.ClientVersion);
+            logger = new IdentityLoggerAdapter(null, Guid.Empty, "comp1", "", false);
+            Assert.AreEqual("comp1", logger.ClientName);
+            logger = new IdentityLoggerAdapter(null, Guid.Empty, "comp1", "version1", false);
+            Assert.AreEqual("comp1", logger.ClientName);
+            Assert.AreEqual("version1", logger.ClientVersion);
         }
 
         [TestMethod()]
-        public void ConstructorComponentTest()
+        public void LegacyLoggerConstructorComponentTest()
         {
-            MsalLogger logger = new MsalLogger(Guid.Empty, null, null, LogLevel.Verbose, false, true, null);
+            ILoggerAdapter logger = new LegacyIdentityLoggerAdapter(Guid.Empty, "", "", LogLevel.Always, false, false, null);
             Assert.AreEqual(string.Empty, logger.ClientName);
             Assert.AreEqual(string.Empty, logger.ClientVersion);
-            Assert.AreEqual(string.Empty, logger.ClientInformation);
-            logger = new MsalLogger(Guid.Empty, "comp1", null, LogLevel.Verbose, false, true, null);
-            Assert.AreEqual(" (comp1)", logger.ClientInformation);
-            logger = new MsalLogger(Guid.Empty, "comp1", "version1", LogLevel.Verbose, false, true, null);
-            Assert.AreEqual(" (comp1: version1)", logger.ClientInformation);
+            logger = new LegacyIdentityLoggerAdapter(Guid.Empty, "comp1", null, LogLevel.Always, false, false, null);
+            Assert.AreEqual("comp1", logger.ClientName);
+            logger = new LegacyIdentityLoggerAdapter(Guid.Empty, "comp1", "version1", LogLevel.Always, false, false, null);
+            Assert.AreEqual("comp1", logger.ClientName);
+            Assert.AreEqual("version1", logger.ClientVersion);
         }
 
         [TestMethod()]
-        [DataRow(LogLevel.Always)]
-        [DataRow(LogLevel.Error)]
-        [DataRow(LogLevel.Warning)]
-        [DataRow(LogLevel.Info)]
-        [DataRow(LogLevel.Verbose)]
-        public void CallbackLoggerTest(LogLevel level)
+        [DataRow(LogLevel.Always, true)]
+        [DataRow(LogLevel.Error, true)]
+        [DataRow(LogLevel.Warning, true)]
+        [DataRow(LogLevel.Info, true)]
+        [DataRow(LogLevel.Verbose, true)]
+        public void CallbackLoggerTest(LogLevel level, bool UseLegaccyLogger)
```

Comment:
Add a similar test for IdentityLogger? Although ExternalMsalLoggerTestAsync kind of covers that?

Was also thinking a test when both WithLogging methods are called - assert that IdentityLogger is used.